pkg/controllers/trusted_ca_bundle_controller.go

```diff
@@ -0,0 +1,216 @@
+package controllers
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/util"
+	"io/ioutil"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	trustedCAConfigMapName      = "ccm-trusted-ca"
+	trustedCABundleConfigMapKey = "ca-bundle.crt"
+	systemTrustBundlePath       = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
+)
+
+type TrustedCABundleReconciler struct {
+	client.Client
+	Scheme          *runtime.Scheme
+	Recorder        record.EventRecorder
+	TargetNamespace string
+	trustBundlePath string
+}
+
+// isSpecTrustedCASet returns true if spec.trustedCA of proxyConfig is set.
+func isSpecTrustedCASet(proxyConfig *configv1.ProxySpec) bool {
+	return len(proxyConfig.TrustedCA.Name) > 0
+}
+
+func (r *TrustedCABundleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	klog.Infof("%s emitted event, syncing %s ConfigMap", req, trustedCAConfigMapName)
+	var CCMTrustedConfigMap *corev1.ConfigMap
```

Comment:
Local variables don't need to be public, also, does this need to be declared here? Or shall we just declare it later on line 74?
```suggestion
	var ccmTrustedConfigMap *corev1.ConfigMap
```